landoapi/transplants.py

```diff
@@ -301,6 +303,43 @@ def warning_wip_commit_message(*, revision, **kwargs):
         return "This revision is marked as a WIP. Please remove `WIP:` before landing."
 
 
+@RevisionWarningCheck(8, "Repository is under a soft code freeze.", True)
+def warning_code_freeze(*, repo, **kwargs):
+    repo_fields = repo.get("fields", {})
+    short_name = repo_fields.get("shortName")
+    if short_name == "mozilla-central":
+        message = ""
+        try:
+            product_details = requests.get(
+                "https://product-details.mozilla.org/1.0/firefox_versions.json"
+            ).json()
+
+            today = datetime.now(tz=timezone.utc)
+            freeze_date = datetime.strptime(
+                f"{product_details.get('NEXT_SOFTFREEZE_DATE')} -0800",
+                "%Y-%m-%d %z",
+            ).replace(tzinfo=timezone.utc)
+            if today < freeze_date:
+                return
+
+            merge_date = datetime.strptime(
+                f"{product_details.get('NEXT_MERGE_DATE')} -0800",
```

Comment:
Is `-0800` relevant in some way? Maybe that should be a variable somewhere and added in as part of the format string.

Reply:
forgot to add a comment in the code - the freeze dates are based on pacific timezone (utc offset of -0800), so this is included to be correctly converted to utc. but agreed it'll look better with a variable